azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -128,40 +122,64 @@ def update_host_plugin_headers(wire_client):
         # Fetching the goal state updates the HostGAPlugin so simply trigger the request
         GoalState._fetch_goal_state(wire_client)
 
-    def update(self, force_update=False):
+    def update(self):
         """
         Updates the current GoalState instance fetching values from the WireServer/HostGAPlugin as needed
         """
-        timestamp = create_timestamp()
-        xml_text, xml_doc, incarnation = GoalState._fetch_goal_state(self._wire_client)
-
-        if force_update or self._incarnation != incarnation:
-            # If we are fetching a new goal state
-            self._history = GoalStateHistory(timestamp, incarnation)
-            self._initialize_basic_properties(xml_doc)
-            self._fetch_extended_goal_state(xml_text, xml_doc, force_vm_settings_update=force_update)
-        else:
-            # else ensure the extensions are using the latest vm_settings
-            timestamp = create_timestamp()
-            vm_settings, vm_settings_updated = self._fetch_vm_settings(force_update=force_update)
+        #
```

Comment:
This block of code implements the fix for the certificate issue. Please review with care,